src/components/sketches/Scene4.ts

```diff
@@ -0,0 +1,170 @@
+      particles[i] = new Particle();
+    }
+
+    flowfield = new Array(rows * cols);
+
+  };
+
+  p.draw = () => {
+    if (showField) {
+      p.background(0);
+    } else {
+      p.background(p.color(0, 0, 0, 5));
+    }
+    let yoff = start;
+    for (let y = 0; y < rows; y++) {
+      let xoff = start;
+      for (let x = 0; x < cols; x++) {
+        const index = x + y * cols;
+        const angle = p.noise(xoff, yoff, zoff) * p.TWO_PI;
+        const v = p5.Vector.fromAngle(angle); // vector from angle
+        const m = p.map(p.noise(xoff, yoff, magOff), 0, 1, -5, 5);
+        v.setMag(m);
+        if (showField) {
+          p.push();
+          p.stroke(255);
+          p.translate(x * SCL, y * SCL);
+          p.rotate(v.heading());
+          const endpoint = p.abs(m) * SCL;
+          p.line(0, 0, endpoint, 0);
+          if (m < 0) {
+            p.stroke('red');
+          } else {
+            p.stroke('green');
+          }
+          p.line(endpoint - 2, 0, endpoint, 0);
+          p.pop();
+        }
+        flowfield[index] = v;
+        xoff += INC;
+      }
+      yoff += INC;
+    }
+    magOff += MAG_INC;
+    zoff += INC_START;
+    start -= MAG_INC;
+
+    if (!showField) {
+      for (let i = 0; i < particles.length; i++) {
+        particles[i].follow(flowfield);
+        particles[i].update();
+        particles[i].edges();
+        particles[i].show();
+      }
+
+      if (p.random(10) > 5 && particles.length < 2500) {
+        const rnd = p.floor(p.noise(zoff) * 20);
+        for (let i = 0; i < rnd; i++) {
+          particles.push(new Particle());
+        }
+      } else if (particles.length > 2000) {
+        const rnd = p.floor(p.random(10));
+        for (let i = 0; i < rnd; i++) {
+          particles.shift();
+        }
+      }
+    }
+  };
+
+  p.windowResized = () => {
+    p.resizeCanvas(window.innerWidth, window.innerHeight, false);
+  };
+
+  class Particle {
+    public pos: p5.Vector;
+    public vel: p5.Vector;
+    private _acc: p5.Vector;
+    private _maxSpeed: number;
+    public _prevPos: p5.Vector;
+    private _bubble: number;
+
+    constructor() {
+      this.pos = p.createVector(p.random(window.innerWidth),p.random(window.innerHeight));
+      this.vel = p.createVector(0, 0);
+      this._acc = p.createVector(0, 0);
+      this._maxSpeed = 2;
+      this._bubble = p.random();
+
+      this._prevPos = this.pos.copy();
+    }
+
+    public update() {
+      this.vel.add(this._acc);
+      this.vel.limit(this._maxSpeed);
+      this.pos.add(this.vel);
+      this._acc.mult(0);
+    }
+
+    public show() {
+      p.strokeWeight(STROKE_WEIGHT);
+      p.line(this.pos.x, this.pos.y, this._prevPos.x, this._prevPos.y);
+      this.updatePrev();
+      //point(this.pos.x, this.pos.y);
+    }
+
+    public updatePrev() {
+      this._prevPos.x = this.pos.x;
+      this._prevPos.y = this.pos.y;
+    }
+
+    public edges() {
+      this.updatePrev();
+    }
```

Comment:
if this only servers the purpoose of doing `this.updatePrev()` lets just call `this.updatePrev()`

Reply:
sounds good